Pending_updates/soc_004a_human_development_index/soc_004a_human_development_index_processing.py

```diff
@@ -0,0 +1,107 @@
+import requests
+import pandas as pd
+import os
+import sys
+import io
+utils_path = os.path.join(os.path.abspath(os.getenv('PROCESSING_DIR')),'utils')
+if utils_path not in sys.path:
+    sys.path.append(utils_path)
+import util_files
+import util_cloud
+import util_carto
+from zipfile import ZipFile
+import logging
+
+# Set up logging
+# Get the top-level logger object
+logger = logging.getLogger()
+for handler in logger.handlers: logger.removeHandler(handler)
+logger.setLevel(logging.INFO)
+# make it print to the console.
+console = logging.StreamHandler()
+logger.addHandler(console)
+logging.basicConfig(format='%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+
+
+# name of table on Carto where you want to upload data
+# this should be a table name that is not currently in use
+dataset_name = 'soc_004a_human_development_index' #check
+
+logger.info('Executing script for dataset: ' + dataset_name)
+# create a new sub-directory within your specified dir called 'data'
+# within this directory, create files to store raw and processed data
+data_dir = util_files.prep_dirs(dataset_name)
+   
+'''
+Download data and save to your data directory
+'''
+logger.info('Downloading raw data')
+# insert the url used to download the data from the source website
+url = 'http://hdr.undp.org/sites/default/files/hdro_statistical_data_tables_1_15_d1_d5.xlsx' #check
+
+# read in data to pandas dataframe
+r = requests.get(url)
+df = pd.read_excel(io.BytesIO(r.content), encoding='utf8')
+
+# save unprocessed source data
+raw_data_file = os.path.join(data_dir, os.path.basename(url))
+df.to_excel(raw_data_file, header=False, index=False)
+
+'''
+Process data
+'''
+# store the information in the metadata rows 
+titles = list(df.iloc[3].fillna(''))
+units = list(df.iloc[4].fillna(''))
+years = list(df.iloc[5].fillna(''))
+headers = []
+for title, year, unit in zip(titles, years, units):
+    headers.append((year + ' ' + title + ' ' + unit).strip())
```

Comment:
```suggestion
# generate headers for the data columns
headers = []
for title, year, unit in zip(titles, years, units):
    # remove any extra whitespace from year
    # add space before year because Carto column name cannot begin with number
    year_edited = ' ' + year.strip()
    # remove any parentheses and extra whitespace from title
    # add space before title because it will follow year
    title_edited = ' ' + title.strip().replace('(', '').replace(')', '')
    # remove any parentheses and extra whitespace from unit
    # replace $ with dollars because Carto cannot have symbols in headers
    # add space before unit because it will follow title
    unit_edited = ' ' + unit.strip().replace('(', '').replace(')', '').replace('$', 'dollars')
    # join year, title, unit into one string and replace spaces with underscores
    headers.append((year_edited + title_edited + unit_edited).strip().replace(' ', '_'))
```